### PR TITLE
[LoongArch64] Fix nonvolatile context restoration referenced 101709.

### PR DESCRIPTION
**Pull request by shushanhf**
Fix nonvolatile context restoration referenced #101709

### CODE REVIEW

File: src/coreclr/pal/src/arch/loongarch64/asmconstants.h [hunk 1]
```diff
@@ -45,8 +45,8 @@
 #define CONTEXT_T6          CONTEXT_T5+SIZEOF_LOONGARCH_GPR
 #define CONTEXT_T7          CONTEXT_T6+SIZEOF_LOONGARCH_GPR
 #define CONTEXT_T8          CONTEXT_T7+SIZEOF_LOONGARCH_GPR
-#define CONTEXT_X0          CONTEXT_T8+SIZEOF_LOONGARCH_GPR
-#define CONTEXT_Fp          CONTEXT_X0+SIZEOF_LOONGARCH_GPR
+#define CONTEXT_R21         CONTEXT_T8+SIZEOF_LOONGARCH_GPR
+#define CONTEXT_Fp          CONTEXT_R21+SIZEOF_LOONGARCH_GPR
```
[the remark this answers is not in the document]
Comment (am11) [reply src/coreclr/pal/src/arch/loongarch64/asmconstants.h:49]: Manual (https://loongson.github.io/LoongArch-Documentation/LoongArch-Vol1-EN.html#registers) suggests r0..r31 which is the scheme llvm is using https://github.com/llvm/llvm-project/blob/d98a78590f4f9e43fdfb69fde7d154a985e4560f/clang/lib/Basic/Targets/LoongArch.cpp#L25. Looks like we are using "GCC aliases" scheme here https://github.com/llvm/llvm-project/blob/d98a78590f4f9e43fdfb69fde7d154a985e4560f/clang/lib/Basic/Targets/LoongArch.cpp#L51.

[the remark this answers is not in the document]
Comment (shushanhf) [reply src/coreclr/pal/src/arch/loongarch64/asmconstants.h:49]: > What's the suggested name for R21 register? Is the alias x0 available in asm?

As the `x0` is misleading to the SIMD registers.  The new spec don't suggest using x0 for R21.
I will delete all related in future.

[the remark this answers is not in the document]
Comment (shushanhf) [reply src/coreclr/pal/src/arch/loongarch64/asmconstants.h:49]: > We are using ABI-ish names in our definition:
> https://github.com/dotnet/runtime/blob/84b33395057737db3ea342a5151feb6b90c1b6f6/src/coreclr/jit/registerloongarch64.h#L40

This is outdated and we didn't use this alias within JIT.
I will delete the others which using `x0` in future.
Thanks

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Pull request merged**